File: android/app/src/main/java/ai/thetahealth/mirobody/data/auth/AuthRepository.kt
package ai.thetahealth.mirobody.data.auth

import ai.thetahealth.mirobody.data.net.unwrap
import ai.thetahealth.mirobody.data.settings.SettingsStore

class AuthRepository(
    private val api: AuthApi,
    private val settings: SettingsStore,
) {
    suspend fun sendCode(email: String) {
        api.sendEmailCode(EmailLoginRequest(email = email)).unwrap()
        settings.setLastEmail(email)
    }

    suspend fun verifyCode(email: String, code: String): AuthTokenResponse {
        val token = api.verifyEmailCode(EmailVerifyRequest(email = email, code = code)).unwrap()
        settings.setAccessToken(token.accessToken)
        settings.setLastEmail(email)
        return token
    }

    suspend fun signOut() {
        settings.clearAuth()
    }
}

File: ios/Mirobody/Data/Auth/AuthRepository.swift
import Foundation

/// Email one-time-code auth. Mirrors `data/auth/AuthRepository.kt`. The Retrofit
/// `AuthApi` interface is folded into direct `ApiClient` calls here.
///
/// Endpoints: `POST /email/login`, `POST /email/verify`.
final class AuthRepository {
    private let api: ApiClient
    private let settings: SettingsStore

    init(api: ApiClient, settings: SettingsStore) {
        self.api = api
        self.settings = settings
    }

    func sendCode(email: String) async throws {
        let _: EmailLoginResponse = try await api.post("/email/login", EmailLoginRequest(email: email))
        settings.setLastEmail(email)
    }

    @discardableResult
    func verifyCode(email: String, code: String) async throws -> AuthTokenResponse {
        let token: AuthTokenResponse = try await api.post(
            "/email/verify",
            EmailVerifyRequest(email: email, code: code)
        )
        settings.setAccessToken(token.accessToken)
        settings.setLastEmail(email)
        return token
    }

    func signOut() {
        settings.clearAuth()
    }
}
